src/Servers/HttpSys/test/FunctionalTests/Listener/Utilities.cs

```diff
@@ -114,8 +114,25 @@ internal static HttpSysListener CreateServerOnExistingQueue(AuthenticationScheme
         internal static async Task<RequestContext> AcceptAsync(this HttpSysListener server, TimeSpan timeout)
         {
             var factory = new TestRequestContextFactory(server);
-            var acceptContext = new AsyncAcceptContext(server, factory);
-            var acceptTask = server.AcceptAsync(acceptContext).AsTask();
+            using var acceptContext = new AsyncAcceptContext(server, factory);
+            
+            async Task<RequestContext> AcceptAsync()
+            {
+                while (true)
+                {
+                    var requestContext = await server.AcceptAsync(acceptContext);
+
+                    if (server.ValidateRequest(requestContext) && server.ValidateAuth(requestContext))
```

Comment:
We should look at combining the two test layers so you don't have to duplicate message pump functionality like this.

Reply:
Yep agree